rust/src/scannerctl/scan_config.rs

```diff
@@ -5,93 +5,87 @@
+    pub quiet: bool,
+
+    /// Path to the feed containing the scan-config.
+    pub path: Option<PathBuf>,
+    /// Path to the XML file with the port list
+    pub portlist: Option<PathBuf>,
+    /// If enabled, parse the scan json from stdin
+    #[clap(short, long)]
+    pub stdin: bool,
+    /// A list of paths of the scan configurations
+    pub scan_config: Vec<PathBuf>,
 }
 
-pub async fn run(root: &clap::ArgMatches) -> Option<Result<(), CliError>> {
-    let (args, _) = crate::get_args_set_logging(root, "scan-config")?;
-
-    let feed = args.get_one::<PathBuf>("path").cloned();
-    let config: Vec<String> = args
-        .get_many::<String>("scan-config")
-        .expect("scan-config is required")
-        .cloned()
-        .collect();
-    let port_list = args.get_one::<String>("portlist").cloned();
+pub async fn run(args: ScanConfigArgs) -> Result<(), CliError> {
+    let port_list = &args.portlist;
     tracing::debug!("port_list: {port_list:?}");
-    let stdin = args.get_one::<bool>("input").cloned().unwrap_or_default();
-    Some(execute(feed.as_ref(), &config, port_list.as_ref(), stdin).await)
+    execute(
+        args.path.as_ref(),
+        &args.scan_config,
+        args.portlist.as_ref(),
+        args.stdin,
+    )
+    .await
 }
 
 async fn execute(
     feed: Option<&PathBuf>,
-    config: &[String],
-    port_list: Option<&String>,
+    config: &[PathBuf],
+    port_list: Option<&PathBuf>,
     stdin: bool,
 ) -> Result<(), CliError> {
-    let map_error = |f: &str, e: Error| CliError {
-        filename: f.to_string(),
-        kind: CliErrorKind::Corrupt(format!("{e:?}")),
-    };
-    let as_bufreader = |f: &str| {
-        let file = std::fs::File::open(f).map_err(|e| CliError {
-            filename: f.to_string(),
-            kind: CliErrorKind::Corrupt(format!("{e:?}")),
-        })?;
+    let map_error =
+        |f: &PathBuf, e: Error| CliErrorKind::Corrupt(format!("{e:?}")).with(Filename(f));
+    let as_bufreader = |f: &PathBuf| {
+        let file = std::fs::File::open(f)
+            .map_err(|e| CliErrorKind::Corrupt(format!("{e:?}")).with(Filename(f)))?;
         let reader = BufReader::new(file);
         Ok::<BufReader<std::fs::File>, CliError>(reader)
     };
     let storage = Arc::new(DefaultDispatcher::new());
-    let mut scan = {
-        if stdin {
-            tracing::debug!("reading scan config from stdin");
-            serde_json::from_reader(std::io::stdin()).map_err(|e| CliError {
-                filename: "".to_string(),
-                kind: CliErrorKind::Corrupt(format!("{e:?}")),
-            })?
-        } else {
-            Scan::default()
-        }
+    let mut scan = if stdin {
+        tracing::debug!("reading scan config from stdin");
+        serde_json::from_reader(std::io::stdin())
+            .map_err(|e| CliErrorKind::Corrupt(format!("{e:?}")))?
+    } else {
+        // TODO: Shouldn't we read it from a file? If so, use FileOrStdin instead of boolean flag
```

Comment:
No, this means that the user doesn't have a predefined scan to be extended by a scan config, therefore returning a default scan is correct.